186170914 XY Plot Attribute Selection

tealefristoe opened the pull request:
PT Story: https://www.pivotaltracker.com/story/show/186170914

kswenson commented on src/plugins/graph/hooks/use-plot.ts:
```diff
@@ -149,7 +147,16 @@ export const usePlotResponders = (props: IPlotResponderProps) => {
+
+  // respond to selection change
+  useEffect(function respondToSelectionChange() {
+    const disposer = reaction(
+      () => [dataset?.selectedAttributeIdString, dataset?.selectedCaseIdString],
+      () => refreshPointSelection()
+    );
+    return () => disposer();
```
FYI, this can be simplified to:
```suggestion
    return reaction(
      () => [dataset?.selectedAttributeIdString, dataset?.selectedCaseIdString],
      () => refreshPointSelection()
    );
```

kswenson reviewed (approved):
👍 Looks good -- surprisingly straightforward

tejal-shah reviewed (commented):
@tealefristoe , this looks good for the most part. Just a couple of observations:
1. When a series that is currently being highlighted is removed from the XY-Plot, shouldn't the highlight jump to the next available series? I'm asking this because when we **add** a series to the xy-plot, we move the highlight to the newly added one and going by the same logic, when we **remove**, shouldn't we move the highlight to the next available one? - https://screenpal.com/watch/c0X3ljVGMRY
2. Would it be possible to retain the green highlight in the legend dropdown when a highlighted series legend is being used? It confused me a little bit when I had a series highlighted and then when I tried to use the legend, the green highlighting on the legend disappeared making me think that I lost my highlighting but then the series remained highlighted in the plot - https://screenpal.com/watch/c0X3llVGMSo
3. Clicking away anywhere in the xy-plot area except in the plot itself doesn't remove the highlight. Shouldn't we be able to remove the highlight by clicking in the legend or axes areas as well? - https://screenpal.com/watch/c0X3lcVGMP7
4. I probably asked this in a different PR but as I was going through zeplin, I found that individual cell selections in the table and datadeck also highlight points in the XY-Plot. This currently does not happen and I'm wondering if this is something that was missed somehow or is perhaps going to be implemented ina  different story/PR? - https://screenpal.com/watch/c0X3lbVGMSH

tealefristoe commented:
@tejal-shah Whenever you're ready for another check, I addressed points 1-3 you brought up.

For 1, I ended up changing it so you no longer highlight the series when you add a new one. This was the easiest way to reach parity with removing a highlighted series, and I think it fits with the PT story as written. It also better follows how adding a column to a table works. It's probably worth considering alternatives for this moving forward, but for now I hope this behavior is ok.

tejal-shah commented:
@tealefristoe , both points  and 2 look fine now. Point 3 still seems to be happening at my end i.e when I click away in an xy-plot that has one or more points highlighted by clicking near the legend area or near the axes, the highlighting doesn't go away. Thi isn't a big deal but I just wanted to let you know that it still happens for me.

tealefristoe commented:
@tejal-shah I see what you mean--clicking the background of the actual graph will deselect everything, but clicking the background of the axes or legend won't. This is how it's worked previously for deselecting cases. I'm not sure why this design choice was made (or if it was intentional or just happened to go this way), but I think it would be appropriate to have a separate story for that behavior change.

tejal-shah reviewed (approved):
Looks good 👍
